timeseries/src/autogluon/timeseries/trainer/abstract_trainer.py

```diff
@@ -577,26 +576,21 @@ def _train_multi(
         logger.info(f"\nStarting training. Start time is {time.strftime('%Y-%m-%d %H:%M:%S')}")
 
         time_start = time.time()
-        if hyperparameters is not None:
-            hyperparameters = copy.deepcopy(hyperparameters)
-        else:
-            if models is None:
-                raise ValueError("Either models or hyperparameters should be provided")
+        hyperparameters = copy.deepcopy(hyperparameters)
 
         if self.save_data and not self.is_data_saved:
             self.save_train_data(train_data)
             if val_data is not None:
                 self.save_val_data(val_data)
             self.is_data_saved = True
 
-        if models is None:
-            models = self.construct_model_templates(
-                hyperparameters=hyperparameters,
-                hyperparameter_tune=hyperparameter_tune_kwargs is not None,  # TODO: remove hyperparameter_tune
-                freq=train_data.freq,
-                multi_window=self.val_splitter.num_val_windows > 0,
-                excluded_model_types=excluded_model_types,
-            )
```

Comment:
Nice, as we get rid of dead branches like these, I feel that we are getting value out of the type checker